### ISSUE TITLE
[DI] Undefined variable: f

### ISSUE THREAD

**Issue by XWB** [symfony/symfony#28710]
**Symfony version(s) affected**: 4.1.6

**Description**  
The Dependency Injection component remains broken in 4.1.6.

**Additional context**  

```
Notice: Undefined variable: f

ErrorException in /cache/dev/ContainerZuWxPUW/srcDevDebugProjectContainer.php (line 1791)

$c = ($this->services['translator'] ?? $this->getTranslatorService());
$d = ($this->privates['HWI\Image\ImageManager'] ?? $this->getImageManagerService());
$e = ($this->services['sonata.intl.templating.helper.datetime'] ?? $this->getSonata_Intl_Templating_Helper_DatetimeService());
$h = ($this->privates['HWI\Helper\UnitHelper'] ?? $this->privates['HWI\Helper\UnitHelper'] = new \HWI\Helper\UnitHelper());

$this->services['hwi_transformer'] = $instance = new \HWI\Bundle\TransformerBundle\Transformer(array('HWI\\Transformer\\Entity\\BrandTransformer' => new \HWI\Transformer\Entity\BrandTransformer($a), 'HWI\\Transformer\\Entity\\CommunityGearTransformer' => new \HWI\Transformer\Entity\CommunityGearTransformer($b, $c), 'HWI\\Transformer\\Entity\\ComparisonTableTransformer' => new \HWI\Transformer\Entity\ComparisonTableTransformer(), 'HWI\\Transformer\\Entity\\MagazineTransformer' => new \HWI\Transformer\Entity\MagazineTransformer($d, $c, $e), 'HWI\\Transformer\\Entity\\NewsTransformer' => new \HWI\Transformer\Entity\NewsTransformer($d, $c, $e), 'HWI\\Transformer\\Entity\\PcAdviceConfigurationTransformer' => new \HWI\Transformer\Entity\PcAdviceConfigurationTransformer($d, $c, $e), 'HWI\\Transformer\\Entity\\ProductGroupTransformer' => new \HWI\Transformer\Entity\ProductGroupTransformer($c), 'HWI\\Transformer\\Entity\\ProductReviewTransformer' => new \HWI\Transformer\Entity\ProductReviewTransformer($d, $c), 'HWI\\Transformer\\Entity\\ProductTransformer' => $f, 'HWI\\Transformer\\Entity\\ReviewTransformer' => new \HWI\Transformer\Entity\ReviewTransformer($d, $c, $e), 'HWI\\Transformer\\Entity\\ShopReviewTransformer' => new \HWI\Transformer\Entity\ShopReviewTransformer($c, $e), 'HWI\\Transformer\\Entity\\SpecTypeBoolTransformer' => new \HWI\Transformer\Entity\SpecTypeBoolTransformer($c, $a), 'HWI\\Transformer\\Entity\\SpecTypeCalcTransformer' => new \HWI\Transformer\Entity\SpecTypeCalcTransformer($c, $g, $h), 'HWI\\Transformer\\Entity\\SpecTypeMcTransformer' => $i, 'HWI\\Transformer\\Entity\\SpecTypeNrTransformer' => new \HWI\Transformer\Entity\SpecTypeNrTransformer($c, $a, $g, $h), 'HWI\\Transformer\\Entity\\SpecTypeTextTransformer' => new \HWI\Transformer\Entity\SpecTypeTextTransformer(), 'HWI\\Transformer\\Entity\\TestConfigTransformer' => new \HWI\Transformer\Entity\TestConfigTransformer($c, $e), 'HWI\\Transformer\\Entity\\VideoTransformer' => new \HWI\Transformer\Entity\VideoTransformer($d, $c, $e), 'HWI\\Transformer\\Entity\\WishListTransformer' => new \HWI\Transformer\Entity\WishListTransformer()), $this->parameters['hwi_transformer.mapping']);

$g = ($this->services['sonata.intl.templating.helper.number'] ?? $this->getSonata_Intl_Templating_Helper_NumberService());

$f = new \HWI\Transformer\Entity\ProductTransformer($b, $g);
$f->setCountry(($this->privates['HWI\Helper\CountryInfo'] ?? $this->getCountryInfoService()));
```

The issue is that `$f` is not initiated above `$this->services['hwi_transformer']`.

**Comment by nicolas-grekas** [comment]
Fixed in #28760

**Comment by XWB** [comment]
@nicolas-grekas I can confirm that #28760 fixes this issue.

**Comment by webmake** [comment]
When this fix will be available on 4.1version?
